src/test/url-loader/multi-url-resolver_test.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+
+import {MultiUrlResolver} from '../../url-loader/multi-url-resolver';
+import {UrlResolver} from '../../url-loader/url-resolver';
+
+abstract class MockResolver implements UrlResolver {
+  canResolveCount: number;
+  resolveCount: number;
+  constructor() {
+    this.resetCounts();
+  }
+
+  resetCounts() {
+    this.canResolveCount = 0;
+    this.resolveCount = 0;
+  }
+  abstract canResolve(url: string): boolean;
+  abstract resolve(url: string): string;
+}
+
+const mockResolverArray = (resolutions: Array<string>) => {
+  return resolutions.map((resolution): MockResolver => {
+    return new class extends MockResolver {
+      constructor() {
+        super();
+      }
+      canResolve() {
+        this.canResolveCount++;
+        return resolution != null;
+      }
+      resolve() {
+        this.resolveCount++;
+        return resolution;
+      }
+    }
+    ();
```

Comment:
oh I see, the `new`... parens around the class expression, or can you do this without subclassing? Just pass resolution to the constructor ofMockResolver